src/Services/EncounterService.php

```diff
@@ -359,7 +359,7 @@ public function insertEncounter($puuid, $data)
         $encounter = generate_id();
         $data['encounter'] = $encounter;
         $data['uuid'] = UuidRegistry::getRegistryForTable(self::ENCOUNTER_TABLE)->createUuid();
-        $data['date'] = date("Y-m-d");
+        if($data['date'] == NULL) $data['date'] = date("Y-m-d");
```

Comment:
Not sure this will pass the style checker.  May need to make it 
```
if($data['date'] == NULL) {
   $data['date'] = date("Y-m-d");
}
```
Another option would be to do this.
```
if(empty($data['date'])) {
   $data['date'] = date("Y-m-d");
}
```